packages/mobile_app/lib/screens/daily/screen/methods/misc.dart

```diff
@@ -3,11 +3,22 @@ part of '../screen.dart';
 extension DailyScreenMiscMethods on _DailyScreenState {
   List<DateTime> get d => widget.day;
   int get selected => widget.currentIndex.wrapped;
-  EnabledProphecies get toShow => sp.appPref.enabledProphecies;
+  EnabledProphecies get toShow => StaticProvider.data.appPref.enabledProphecies;
   bool get isToday => d[selected].millisecondsSinceEpoch == dtDay;
 
   void calculateProphecy() {
-    sp.prophecyBloc.add(CalculateProphecy(d[selected].millisecondsSinceEpoch,
-        isDebug: sp.debug.isDebug));
+    StaticProvider.prophecyBloc.add(CalculateProphecy(
+        d[selected].millisecondsSinceEpoch,
+        isDebug: StaticProvider.debug.isDebug));
+  }
+
+  Future<bool> internetCheck() async {
+    try {
+      final result = await InternetAddress.lookup('pub.dev');
```

Comment:
Quick and dirty :) 
A nice way of checking -- rely on phone's connectivity API. Im sure there is a package .

I would suggest to change this to something more relaible like `google.com`

Reply:
google.com can be banned in some countries and pub.dev - no